src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/StreamPipeReader.cs

```diff
@@ -525,8 +525,8 @@ private bool TryReadInternal(CancellationTokenSource source, out ReadResult resu
 
         private ReadOnlySequence<byte> GetCurrentReadOnlySequence()
         {
-            Debug.Assert(_readHead != null && _readTail != null);
-            return new ReadOnlySequence<byte>(_readHead, _readIndex, _readTail, _readTail.End);
+            // If _readHead is null then _readTail is also null
```

Comment:
nit: could add a debug.assert for that and then remove the `!`

Reply:
Debug.Assert hid the failure in the first place 😄. I guess this is no better